ls1intum/Artemis
# [Enhancement] Add audit event logs for grading actions

### Checklist

### Motivation and Context



Better logging.

### Description


Added audit logs for 
- Re-Evaluate
- Trigger-All
- Reset Grading

### Steps for Testing


**A:** Test on the test-server, that the three above actions still work.

1. Log in to Artemis
2. Navigate to the grading page of a programming exercise
3. Test the 'Re-Evaluate' button and check the updated results
3. Test the 'Trigger-All' button and check the new results
4. Test the 'Reset' button (works only for test cases)

**B:** Test locally, that for each of the above actions an audit log appears in the database

1. Do the same steps as in **A**
2. Open the _jhi_persistent_audit_event_ table for your local Artemis database
3. Check that for each action an audit event exists (ReEvaluate, TriggerAll, Reset)
4. Open the _jhi_persistent_audit_evt_data_ table for your local Artemis database
5. Check that each audit event has the correct details (exercise, course, etc.)

### Test Coverage


Nothing changed.

### Screenshots



example _jhi_persistent_audit_event_ table
<img width="552" alt="Bildschirmfoto 2020-12-03 um 11 41 24" src="https://user-images.githubusercontent.com/13920539/100999255-d151e280-355c-11eb-93c9-7001236c87fd.png">
example _jhi_persistent_audit_evt_data_ table
<img width="458" alt="Bildschirmfoto 2020-12-03 um 11 41 37" src="https://user-images.githubusercontent.com/13920539/100999276-dca50e00-355c-11eb-8056-d86dd9a79df5.png">
